tensorflow/lite/micro/kernels/xtensa/sub.cc

```diff
@@ -104,11 +148,54 @@ TfLiteStatus EvalSubQuantized(TfLiteContext* context, TfLiteNode* node,
             tflite::micro::GetTensorShape(output),
             tflite::micro::GetTensorData<int8_t>(output));
       }
+#endif  // defined(HIFI4)
       break;
     }
     case kTfLiteInt16: {
+#if defined(HIFI4)
+      int err;
+      const RuntimeShape extended_input1_shape =
+          RuntimeShape::ExtendedShape(5, tflite::micro::GetTensorShape(input1));
+      const RuntimeShape extended_input2_shape =
+          RuntimeShape::ExtendedShape(5, tflite::micro::GetTensorShape(input2));
+      const RuntimeShape extended_output_shape =
+          RuntimeShape::ExtendedShape(5, tflite::micro::GetTensorShape(output));
+      const int* input1_dims = extended_input1_shape.DimsData();
+      const int* input2_dims = extended_input2_shape.DimsData();
+      const int* output_dims = extended_output_shape.DimsData();
+      int b;
+      int inp1_off = 0;
+      int inp2_off = 0;
+      int out_off;
+      out_off =
```

Comment:
we should be naming these variables better. Let's add that to the cleanup/refactor bug and address in a follow-up PR.

Reply:
Added a new bug https://b.corp.google.com/issues/259724572